### ISSUE TITLE
how to realize the custom weight decay?

### ISSUE THREAD

**Issue by huoshuai-dot** [open-mmlab/mmclassification#526]
I want to realize a function (custom weight decay),but I have no idea about it?

**Comment by mzr1996** [comment]
What kind of "custom"? Apply different weight decay ratios to some specific weights or use a new weight decay method (instead of L2 penalty)

**Comment by huoshuai-dot** [comment]
> What kind of "custom"? Apply different weight decay ratios to some specific weights or use a new weight decay method (instead of L2 penalty)

to use a new weight decay method

**Comment by mzr1996** [comment]
> > What kind of "custom"? Apply different weight decay ratios to some specific weights or use a new weight decay method (instead of L2 penalty)
> 
> to use a new weight decay method

Maybe you can create a custom loss and add your weight decay penalty item to that loss.
Otherwise, create a new optimizer and implement your weight decay method in it. You can refer to [the implementation of SDG](https://github.com/pytorch/pytorch/blob/f9422e1c6b5dab15eacbbb0e1d4370b07db1e8e8/torch/optim/_functional.py#L174-L175).

**Comment by huoshuai-dot** [comment]
> > > What kind of "custom"? Apply different weight decay ratios to some specific weights or use a new weight decay method (instead of L2 penalty)
> > 
> > 
> > to use a new weight decay method
> 
> Maybe you can create a custom loss and add your weight decay penalty item to that loss. Otherwise, create a new optimizer and implement your weight decay method in it. You can refer to [the implementation of SDG](https://github.com/pytorch/pytorch/blob/f9422e1c6b5dab15eacbbb0e1d4370b07db1e8e8/torch/optim/_functional.py#L174-L175).

ok! I will try it! 3q!
